requests/utils.py

```diff
@@ -358,13 +358,12 @@ def get_encoding_from_headers(headers):
 
 def stream_decode_response_unicode(iterator, r):
     """Stream decodes a iterator."""
+    encoding = r.encoding
 
     if r.encoding is None:
-        for item in iterator:
-            yield item
-        return
+        encoding = r.apparent_encoding
 
-    decoder = codecs.getincrementaldecoder(r.encoding)(errors='replace')
+    decoder = codecs.getincrementaldecoder(encoding)(errors='replace')
```

Comment:
Can we add the same `try...except` as in `text, except catching both `LookupError` *and* `TypeError` and, if either is raised, raising a more appropriate error? This code cannot function with a fallback, so we need to raise a useful exception here.